uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-requests_test.js

```diff
@@ -1,30 +1,899 @@
+
+      describe('filterLeaveRequestByStatus', function () {
+
+        var returnValue;
+
+        describe('when status is blank', function () {
+
+          beforeEach(function () {
+            returnValue = controller.filterLeaveRequestByStatus('');
+          });
+
+          it('returns all data', function () {
+            expect(returnValue).toEqual(controller.leaveRequests.filter.list);
+          });
+        });
+
+        describe('when status is all', function () {
+
+          beforeEach(function () {
+            returnValue = controller.filterLeaveRequestByStatus({
+              name: 'all'
+            });
+          });
+
+          it('returns all data', function () {
+            expect(returnValue).toEqual(controller.leaveRequests.filter.list);
+          });
+        });
+
+        describe('for any other status', function () {
+
+          var status,
+            expectedValue;
+
+          beforeEach(function () {
+            status = optionGroupMock.getCollection('hrleaveandabsences_leave_request_status')[0];
+            returnValue = controller.filterLeaveRequestByStatus(status);
+            expectedValue = controller.leaveRequests.filter.list.filter(function (request) {
+              return request.status_id == status.value;
+            });
+          });
+
+          it('returns all data', function () {
+            expect(returnValue).toEqual(expectedValue);
+          });
+        });
+      });
+
+      describe('getNavBadge::', function () {
+
+        var returnValue;
+
+        describe('when status is approved', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('approved')
+          });
+
+          it('returns badge-success', function () {
+            expect(returnValue).toBe('badge-success');
+          })
+        });
+
+        describe('when status is rejected', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('rejected')
+          });
+
+          it('returns badge-danger', function () {
+            expect(returnValue).toBe('badge-danger');
+          })
+        });
+
+        describe('when status is cancelled', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('cancelled')
+          });
+
+          it('returns blank string', function () {
+            expect(returnValue).toBe('');
+          })
+        });
+
+        describe('when status is all', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge('all')
+          });
+
+          it('returns blank string', function () {
+            expect(returnValue).toBe('');
+          })
+        });
+
+        describe('when status is some other value', function () {
+
+          beforeEach(function () {
+            returnValue = controller.getNavBadge(jasmine.any(String))
+          });
+
+          it('returns blank string', function () {
```

Comment:
wrong title

Please double-check your code before sending it to review, these are obvious copy-paste errors